src/services/reviews.js

```diff
@@ -0,0 +1,534 @@
+			if (isPublishResource) {
+				const publishResource = await resourceService.publishResource(
+					resourceId,
+					validateReview.resource.user_id
+				)
+				return publishResource
+			}
+			// updating the resource last reviewed at
+			await resourceQueries.updateOne(
+				{ organization_id: validateReview.resource.organization_id, id: resourceId },
+				{ last_reviewed_on: new Date() }
+			)
+
+			return responses.successResponse({
+				statusCode: httpStatusCode.ok,
+				message: 'REVIEW_UPDATED',
+			})
+		} catch (error) {
+			throw error
+		}
+	}
+
+	/**
+	 * Report or Reject resource
+	 * @method
+	 * @name rejectOrReportResource
+	 * @param {Object} bodyData - review body data.
+	 * @param {Integer} resourceId - resource id.
+	 * @param {Boolean} isReported - Indicate the resource is reported or not
+	 * @param {String} userId - logged in user id.
+	 * @param {String} orgId - organization id.
+	 * @returns {JSON} - review approved response.
+	 */
+
+	static async rejectOrReportResource(resourceId, isReported, bodyData, userId, orgId) {
+		try {
+			//validate resource
+			let validateReview = await this.validateReview(resourceId, userId, orgId)
+			if (validateReview.statusCode !== httpStatusCode.ok) {
+				return validateReview
+			}
+
+			validateReview = validateReview.result
+
+			//program cannot reject or report
+			if (validateReview.resource.type === common.RESOURCE_TYPE_PROGRAM) {
+				return responses.failureResponse({
+					message: 'PROGRAM_REJECTION_NOT_ALLOWED',
+					statusCode: httpStatusCode.bad_request,
+					responseCode: 'CLIENT_ERROR',
+				})
+			}
+
+			// Add or update comments
+			if (bodyData.comment) {
+				await handleComments(bodyData.comment, resourceId, userId)
+			}
+
+			let updateObj = {
+				status: isReported ? common.REVIEW_STATUS_REJECTED_AND_REPORTED : common.REVIEW_STATUS_REJECTED,
+			}
+
+			if (bodyData.notes) {
+				updateObj.notes = bodyData.notes
+			}
+
+			//update reviews
+			await reviewsQueries.update(
+				{ id: validateReview.review.id, organization_id: validateReview.review.organization_id },
+				updateObj
+			)
+
+			updateObj.last_reviewed_on = new Date()
+
+			//update the resource
+			await resourceQueries.updateOne(
+				{ id: resourceId, organization_id: validateReview.resource.organization_id },
+				_.omit(updateObj, ['notes'])
+			)
+
+			const message = isReported ? 'REVIEW_REJECTED_AND_REPORTED' : 'REVIEW_REJECTED'
+
+			return responses.successResponse({
+				statusCode: httpStatusCode.ok,
+				message,
+			})
+		} catch (error) {
+			throw error
+		}
+	}
+
+	/**
+	 * Validating the Resource for review creation or update
+	 * @method
+	 * @name validateReview
+	 * @param {Integer} resourceId - resource id.
+	 * @param {String} userId - logged in user id.
+	 * @returns {JSON} - review updated response.
+	 */
+
+	static async validateReview(resourceId, userId) {
+		try {
+			//get resource details
+			let resourceDetails = await resourceService.getDetails(resourceId)
```

Comment:
@priyanka-TL All these various blocks should be different functions, each will take only the necessary resource information. The calling function will call these functions with relevant information as needed and handle the responses accordingly.